### PR TITLE
ssdfs-utils: init at 4.27

### PR DESCRIPTION
**Pull request by ners**
###### Description of changes

This PR adds tools for SSDFS, a new filesystem optimised for NVME SSDs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/ssdfs-utils/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+stdenv.mkDerivation {
+  pname = "ssdfs-utils";
+  version = "4.24";
+
+  src = fetchFromGitHub {
+      owner = "dubeyko";
+      repo = "ssdfs-tools";
+      rev = "0a3ca85b454e56cf2422f33663639ad13fc792dd";
+      hash = "sha256-pzpwbvo0BR4fG6Cnxb9GdH/UZlJ6C5am7TULSLEzdvI=";
```
Comment (alyssais) [inline pkgs/tools/filesystems/ssdfs-utils/default.nix:19]: In Nixpkgs we use two spaces for indentation of Nix code.

File: pkgs/tools/filesystems/ssdfs-utils/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  buildInputs = [
+    autoconf
+    automake
+    libtool
+    libuuid
+    zlib
+  ];
+
+  preConfigure = "./autogen.sh";
```
Comment (alyssais) [inline pkgs/tools/filesystems/ssdfs-utils/default.nix:30]: Does `autoreconfHook` work instead of this?

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -4050,6 +4050,8 @@ with pkgs;
 
   btrfs-snap = callPackage ../tools/filesystems/btrfs-snap { };
 
+  ssdfs-utils = callPackage ../tools/filesystems/ssdfs-utils { };
```
Comment (alyssais) [inline pkgs/top-level/all-packages.nix:4053]: Why not ssdfs-tools, like the repo (and also the alpine package)?

**Review by alyssais — CHANGES_REQUESTED** [review]
Almost there!

File: pkgs/tools/filesystems/ssdfs-utils/default.nix (hunk 1 above)
Comment (ners) [reply pkgs/tools/filesystems/ssdfs-utils/default.nix]: Whoops, somehow the autoformatter didn't fire. :sweat_smile:

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (ners) [reply pkgs/top-level/all-packages.nix]: Good catch. I also wasn't sure which of the two to use. I ended up going with what the upstream author seems to intend:
 - commit messages reference `ssdfs-utils`: https://github.com/dubeyko/ssdfs-tools/commits/master
 - the name of the autoconf project is also `ssdfs-utils`: https://github.com/dubeyko/ssdfs-tools/blob/master/configure.ac#L6

File: pkgs/tools/filesystems/ssdfs-utils/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner = "dubeyko";
+    repo = "ssdfs-tools";
+    rev = "0a3ca85b454e56cf2422f33663639ad13fc792dd";
+    hash = "sha256-pzpwbvo0BR4fG6Cnxb9GdH/UZlJ6C5am7TULSLEzdvI=";
+  };
+
+  buildInputs = [
+    autoreconfHook
```
Comment (alyssais) [inline pkgs/tools/filesystems/ssdfs-utils/default.nix:22]: This needs to be in `nativeBuildInputs`.  Set `strictDeps` or try cross compiling to make sure you have the dependencies correctly categorised.

**Review by alyssais — CHANGES_REQUESTED** [review]

File: pkgs/tools/filesystems/ssdfs-utils/default.nix (hunk 4 above)
Comment (ners) [reply pkgs/tools/filesystems/ssdfs-utils/default.nix:22]: Thank you for the excellent review and helpful explanation.

**Comment by alyssais** [comment]
@OfBorg build ssdfs pkgsCross.riscv64.ssdfs

**Review by alyssais — APPROVED** [review]

**Pull request merged**